iwyu_test_util.py

```diff
@@ -419,6 +436,112 @@ def _GetLaunchArguments(cc_file):
   return shlex.split(args)
 
 
+def _CheckForCompileTimeMacro(macro):
+    """This assumes that the local clang++ and the include-what-you-use
+    under test are using the same libclang, and therefore the same default libraries.
+
+    Look for the specified macro in the clang output.
+    """
+    txt = '#include <cstddef>'
+
+    result = subprocess.run(['clang', '-dM', '-E', '-x', 'c++', '-'], capture_output=True,
+                            input=txt, text=True)
+
+    return " "+macro+" " in str(result.stdout)
+
+
+def _CheckExtraCommandLine(args):
+  """Check if particular options are on the command line"""
+  env_iwyu_extra_args = os.getenv('IWYU_EXTRA_ARGS')
+  iwyu_extra_args = shlex.split(env_iwyu_extra_args)
+  standardised_args = ' '.join(iwyu_extra_args)
+  return args in standardised_args
+
+
+def _EvalCondition(cond):
+  """Evaluate the input condition.
+
+  All conditions comprise a label and a single argument. The supported
+  tests are:
+
+  IwyuHasTargetSupportFor arg - whether include-what-you-use supports
+                                the architecture 'arg' as a target.
+
+  TestingCXXSTL (libc++|libstdcpp|canonical) - whether the current
+                                test is compiling against a particular
+                                C++ standard library.
+  """
+  # Some helper functions to handle conditions
+  def isTestingLibCxx():
+    return _CheckExtraCommandLine('-stdlib=libc++') or _CheckForCompileTimeMacro('_LIBCPP_VERSION')
+  def isTestingLibstdcpp():
+    return _CheckExtraCommandLine('-stdlib=libstdc++') or _CheckForCompileTimeMacro('__GLIBCXX__')
+  def isTestingCanonicalClangMappings():
+    return _CheckExtraCommandLine('-Xiwyu --experimental=clang_mappings')
+  def iwyuHasTargetSupportFor(tgt):
+    iwyu = _GetIwyuPath()
+    result = subprocess.run([iwyu, '--print-targets'], capture_output=True)
```

Comment:
On my system, this prints:
```
kimgr@lardo:~/code/iwyu/include-what-you-use$ include-what-you-use -print-targets

  Registered Targets:
    aarch64     - AArch64 (little endian)
    aarch64_32  - AArch64 (little endian ILP32)
    aarch64_be  - AArch64 (big endian)
    amdgcn      - AMD GCN GPUs
    arm         - ARM
    arm64       - ARM64 (little endian)
    arm64_32    - ARM64 (little endian ILP32)
    armeb       - ARM (big endian)
    avr         - Atmel AVR Microcontroller
    bpf         - BPF (host endian)
    bpfeb       - BPF (big endian)
    bpfel       - BPF (little endian)
    hexagon     - Hexagon
    lanai       - Lanai
    loongarch32 - 32-bit LoongArch
    loongarch64 - 64-bit LoongArch
    m68k        - Motorola 68000 family
    mips        - MIPS (32-bit big endian)
    mips64      - MIPS (64-bit big endian)
    mips64el    - MIPS (64-bit little endian)
    mipsel      - MIPS (32-bit little endian)
    msp430      - MSP430 [experimental]
    nvptx       - NVIDIA PTX 32-bit
    nvptx64     - NVIDIA PTX 64-bit
    ppc32       - PowerPC 32
    ppc32le     - PowerPC 32 LE
    ppc64       - PowerPC 64
    ppc64le     - PowerPC 64 LE
    r600        - AMD GPUs HD2XXX-HD6XXX
    riscv32     - 32-bit RISC-V
    riscv64     - 64-bit RISC-V
    sparc       - Sparc
    sparcel     - Sparc LE
    sparcv9     - Sparc V9
    systemz     - SystemZ
    thumb       - Thumb
    thumbeb     - Thumb (big endian)
    ve          - VE
    wasm32      - WebAssembly 32-bit
    wasm64      - WebAssembly 64-bit
    x86         - 32-bit X86: Pentium-Pro and above
    x86-64      - 64-bit X86: EM64T and AMD64
    xcore       - XCore
    xtensa      - Xtensa 32
```

There's quite a bit of substring-ambiguity there, e.g. `x86`/`x86-64`, `mips`/`mips64`/`mips64el`/`mipsel`, etc.

It might be worth doing a little bit more rigorous parsing at some point (not a priority right now!)